### PR TITLE
feat: supports plugin drag and plugin order state is persisted using …

### PR DESCRIPTION
**Pull request by STATICHIT**
…vueuse

English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution
1.支持左上，左下，右上三个位置的插件内部/跨越拖拽。
2.对插件顺序进行了持久化保存。

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced drag-and-drop functionality for rearranging plugins in the layout.
	- Added a new `fixed` position option for enhanced layout configuration.
	- Enhanced navigation panels with draggable components for dynamic item reordering.

- **Improvements**
	- Improved plugin management through systematic alignment and indexing.
	- Streamlined user interaction in the navigation and settings components.

- **Dependencies**
	- Added `vue-draggable-next` to enhance drag-and-drop capabilities.

### CODE REVIEW

**Review by lizhijie429 — APPROVED** [review]

File: packages/controller/src/useLayout.js [hunk 1]
```diff
@@ -198,6 +203,61 @@ export default () => {
     }
   }
 
+  //拖拽后改变插件位置
+  const dargPluginLayout = (from, to, oldIndex, newIndex) => {
```
Comment (hexqi) [inline packages/controller/src/useLayout.js]: 拼写错误： --> drag

File: packages/controller/src/useLayout.js [hunk 2]
```diff
@@ -198,6 +203,61 @@ export default () => {
...
+    if (movedItem) {
+      movedItem.align = to
+      movedItem.index = newIndex
+    }
+  }
+
+  //判断是否在同一侧
+  const isSameSide = (from, to) => {
+    const leftSide = ['leftTop', 'leftBottom']
+    const rightSide = ['rightTop', 'rightBottom']
```
Comment (hexqi) [inline packages/controller/src/useLayout.js]: 上面已经定义了常量了，这里可以直接引用常量PLUGIN_POSITION.xxx

File: packages/design-core/package.json [hunk 3]
```diff
@@ -94,6 +94,7 @@
     "prettier": "2.7.1",
     "sortablejs": "^1.14.0",
     "vue": "3.4.23",
+    "vue-draggable-plus": "^0.5.3",
```
Comment (hexqi) [inline packages/design-core/package.json]: 项目中有用到vue-draggable-next，看下vue-draggable-next能否满足要求，统一下依赖

File: packages/design-core/src/App.vue [hunk 4]
```diff
@@ -75,10 +75,32 @@ export default {
       jsClose: null
     })
 
+    // Step 1: 收集插件的 align 信息
+    const alignGroups = {}
+    const pluginList = addons.plugins
+
+    pluginList.forEach((item) => {
+      if (item.id) {
+        const align = item.options?.align || 'rightTop'
```
Comment (hexqi) [inline packages/design-core/src/App.vue]: 默认在左上更好点

File: packages/design-core/src/DesignSettings.vue [hunk 5]
```diff
@@ -18,29 +18,23 @@
...
-          'list-item': true,
-          'first-item': index === 0,
-          active: item.id === renderPanel
-        }"
+    <VueDraggable id="rightTop" v-model="state.leftList" class="nav-panel-lists" group="plugins" @end="onEnd">
+      <div
+        v-for="item in state.leftList"
+        :key="item.id"
+        class="list-item"
+        :class="{ 'first-item': item === state.leftList[0], active: item.id === renderPanel }"
```
Comment (hexqi) [inline packages/design-core/src/DesignSettings.vue]: 27、28行可以合并，`:class="['list-item', { 'first-item': item === state.leftList[0], active: item.id === renderPanel }]"`

**Review by hexqi — APPROVED** [review]

**Pull request merged**